### PR TITLE
Cortex XDR CCP content creation

### PR DESCRIPTION
**Pull request by ltipp147**
Required items, please complete
   
   Change(s):
   - Added content structure for Sentinel solution - Cortex XDR (ONE Data Connector and THREE Analytic Rules).
   - Added Cortex XDR Logo in the folder \Azure-Sentinel\Logos
   - Added sample data in the folder Azure-Sentinel\Sample Data\Custom
   - Added custom logs table schema in the folder Azure-Sentinel\.script\tests\KqlvalidationsTests\CustomTables

   Reason for Change(s):
   - Initial commit

   Version Updated:
   - N/A

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by ltipp147** [comment]
>

**Comment by ltipp147** [comment]
@microsoft-github-policy-service agree company="DEFEND Ltd."

L Tippayaratprontawee | Cybersecurity Analyst | +64 211 412 122 | DEFEND

**Comment by v-prasadboke** [comment]
Hello @ltipp147 please repackage the solution. 
Please add solutionmetadata and Input file in the solution folder.
To repackage the solution please go through this Readme.md file
https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V2/README.md

You can refer to any of the solution folder for folder structure

**Comment by v-prasadboke** [comment]
Hello @ltipp147 any updates on the above

**Comment by ltipp147** [comment]
Hi @v-prasadboke, apologies for late response - we're working on it and will add the package within this week, thanks!

**Comment by ltipp147** [comment]
Hi @v-prasadboke the solution has been repackaged with SolutionMetadata and Input file in the solution folder. Please, any feedback is appreciated.

File: Solutions/Cortex XDR/SolutionMetadata.json
Comment (v-prasadboke) [inline Solutions/Cortex XDR/SolutionMetadata.json:1]: Hello @ltipp147 solution metadata should be in solution folder, not in package.
Package folder should contain maintemplate and createui and a zip package

**Comment by v-prasadboke** [comment]
Please update the branch from master so that i can pull it my local

**Comment by v-prasadboke** [comment]
Hello @ltipp147 can you please specify the sampledata in which analytic rule is being used

**Comment by ltipp147** [comment]
Hi @v-prasadboke, I believe I have satisfied the requested requirements. Please see the screenshot below for the folder structure and sample data that have already been uploaded:

- The SolutionMetadata.json is in the Solution (Cortex XDR) folder
- The Package folder contains (1) mainTemplate.json, (2) createUiDefinition.json, and (3) a zip package of version 3.0.0 (generated by createSolutionV3.ps1 toolkit)
- The sample data in which analytic rule is being used has been uploaded in the _Sample Data/Custom_ folder; these include (1) Raw Logs, (2) Ingested Logs, and (3) Schema.

CortexXDRSolution

**Comment by v-prasadboke** [comment]
Thank you for the clarification @ltipp147, Will update you shortly on this

**Comment by v-prasadboke** [comment]
Hello @ltipp147 it seems, this solution have a data connector and analytic rules are dependent on it, i request you to fill the data mentioned in all the analytic rules. 
Data to be filled :- requiredDataConnectors, data types, tactics: [] & relevantTechniques: [],

You can refer this Detection for more clarification 
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Network%20Session%20Essentials/Analytic%20Rules/AnomalyFoundInNetworkSessionTraffic.yaml

**Comment by ltipp147** [comment]
Hi @v-prasadboke, I have updated the solution as requested:
- Added requiredDataConnectors and dataTypes to all analytic rules
- Added data connector ID 'CortexXDR' to the list in 'ValidConnectorIds.json' file to fix DetectionTemplateSchemaValidation fail

Let me know if more detail is required, thanks!

**Comment by v-prasadboke** [comment]
Hello @ltipp147, Thank you for committing the changes.
Will be investigating this PR, you can expect an update till 25 Jul 2023

**Comment by v-prasadboke** [comment]
Hello @ltipp147 Still data fields are missing for tactics: [] , relevantTechniques: [] from all the analytic rules

File: Solutions/Cortex XDR/Analytic Rules/CortexXDR_High.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  - connectorId: CortexXDR
+    dataTypes:
+      - CortexXDR_Incidents_CL
+queryFrequency: 5m
+queryPeriod: 5m
+triggerOperator: gt
+triggerThreshold: 0
+suppressionDuration: 5h
+suppressionEnabled: false
+tactics: []
```
Comment (v-prasadboke) [inline Solutions/Cortex XDR/Analytic Rules/CortexXDR_High.yaml:16]: This field is missing

Comment (ltipp147) [reply Solutions/Cortex XDR/Analytic Rules/CortexXDR_High.yaml]: Hi @v-prasadboke, tactics have been updated for all three analytic rules as requested

**Comment by v-prasadboke** [comment]
Thank you @ltipp147 for committing the changes, you can expect an update till 26 Jul 2023

**Comment by v-prasadboke** [comment]
Hello @ltipp147 you still havent filled relevant techniques of all the analytic rules.
Can you please add the required fields.

**Comment by dark-binary** [comment]
Hey @v-prasadboke , are tactics and technique a mandatory field for the solution? The reason I ask is because this solution aims to pull incidents from Cortex XDR (and each incident might come up with different tactic and technique mapped with it). We can add a generic technique that MIGHT be suitable for all incidents, but would be on a high level rather than a targeted mapping. Would prefer leaving out the tactics and/or technique field empty if it isn't mandatory, or we are happy to add a generic mapping of tactic and technique if that will help pass the solution.

**Comment by v-prasadboke** [comment]
Hello @dark-binary, I got the issue you are facing. Just give me some time, Will update about this within a short span of time.

**Comment by v-prasadboke** [comment]
Hello @dark-binary, you can keep it empty or we have another option of dynamic mapping of techniques and tactics.

You can go through this link for customizing the alert.
https://learn.microsoft.com/en-us/azure/sentinel/customize-alert-details

**Comment by dark-binary** [comment]
Hey @v-prasadboke , it would be best kept empty for the moment. We will work on dynamic mapping as an improvement, since the values from the alert cannot be directly applied for the fields (needs to be sanitized). 

Can we skip the tactics and technique mapping for now, and I can have our dev team work on the dynamic mapping from the alert as an improvement later, if that works?

**Comment by v-prasadboke** [comment]
Ok sure @dark-binary, ill continue with the same and update you till EOD.

**Comment by v-prasadboke** [comment]
Hello @dark-binary can you please add release notes in the PR.
Guidance to release notes - https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/ReleaseNotesGuidance.md

And also update the zip package with latest createui and maintemplate. 
Createui and maintemplate in the zip should match with createui and maintemplate outside the zip

**Comment by ltipp147** [comment]
Hey @v-prasadboke the latest pull request has the below updates:

1 - Removed tactics from the 3 analytic rules (these will be added along with techniques as an improvement in the next version)
2 - Added ReleaseNotes.md
3 - Updated the zip package to reflect the latest changes in maintemplate and createui files

Hopefully everything is good. Please let us know if you have any questions.

**Comment by v-prasadboke** [comment]
Thank you for updating @ltipp147, Will have a look at this and let you know till EOD.

**Comment by ltipp147** [comment]
Hi @v-prasadboke, apologize for the oversight in updating the files. The necessary updates have now been completed to reflect the most current information:

1 - Added the ReleaseNotes hyperlink to the CreateUiDefinition.json file
2 - Updated the zip package to match the mainTemplate and CreateUiDefinition files in and outside the zip folder

Please let us know if you have any further questions regarding the updated files. Thank you!

**Review by v-prasadboke — APPROVED** [review]
All looks good

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**